dnf/base.py

```diff
@@ -1005,7 +1006,7 @@ def _run_transaction(self, cb):
         else:
             if self._record_history():
                 herrors = [ucd(x) for x in errors]
-                self.history.end(rpmdbv, 2, errors=herrors)
+                self.history.end(rpmdbv)
```

Comment:
I believe here is a problem. First I would like to move self.history.end(rpmdbv) few lines bellow. Also I believe that in this case it is normal that some elements were not done. 

```
        else:
            if self._record_history():
                herrors = [ucd(x) for x in errors]
                self.history.end(rpmdbv)

            logger.critical(_("Transaction couldn't start:"))
            for e in errors:
                logger.critical(ucd(e[0]))  # should this be 'to_unicoded'?

            if self._record_history():
                herrors = [ucd(x) for x in errors]
                self.history.end(rpmdbv)
            msg = _("Could not run transaction.")
            raise dnf.exceptions.Error(msg)

```